pkg/dogstatsd/server_worker.go

```diff
@@ -22,13 +31,20 @@ func (w *worker) flush() {
 }
 
 func (w *worker) run() {
+	// we allocate it once for the life of the worker instead of once per
+	// packet. This will be use to store the samples out a of packets.
+	// Allocating it every time is very costly, especially on the GC.
+	samples := make([]metrics.MetricSample, 0, defaultSampleSize)
```

Comment:
That's a nit but to be more explicit on its lifecycle ("once for a worker"), it could be part of `struct worker`.

Reply:
yes ! I started this PR before the `stuct worker` was introduce. Good catch